dependency/health_service.go

```diff
@@ -122,18 +124,25 @@ func healthServiceQuery(s string, connect bool) (*HealthServiceQuery, error) {
 		return nil, err
 	}
 
-	return &HealthServiceQuery{
-		stopCh:    make(chan struct{}, 1),
-		dc:        m["dc"],
-		filters:   filters,
-		name:      m["name"],
-		near:      m["near"],
-		tag:       m["tag"],
-		connect:   connect,
-		namespace: queryParams.Get(QueryNamespace),
-		peer:      queryParams.Get(QueryPeer),
-		partition: queryParams.Get(QueryPartition),
-	}, nil
+	qry := &HealthServiceQuery{
+		stopCh:        make(chan struct{}, 1),
+		dc:            m["dc"],
+		filters:       filters,
+		name:          m["name"],
+		near:          m["near"],
+		tag:           m["tag"],
+		connect:       connect,
+		namespace:     queryParams.Get(QueryNamespace),
+		peer:          queryParams.Get(QueryPeer),
+		partition:     queryParams.Get(QueryPartition),
+		samenessGroup: queryParams.Get(QuerySamenessGroup),
+	}
+
+	if qry.samenessGroup != "" && qry.peer != "" {
+		return nil, fmt.Errorf("health.service: cannot specify both peer and sameness-group")
+	}
```

Comment:
nit/personal preference/feel free to ignore: to me it makes more sense to do the validation _before_ constructing the object.

Reply:
good call. thank you.

Reply:
elevated the guard clause in edd0f7394427f64c6fc9de7aa74dd6bac9c32f80